fdbserver/DataDistributorInterface.h

```diff
@@ -36,6 +36,7 @@ struct DataDistributorInterface {
 	RequestStream<struct DistributorExclusionSafetyCheckRequest> distributorExclCheckReq;
 	RequestStream<struct GetDataDistributorMetricsRequest> dataDistributorMetrics;
 	RequestStream<struct DistributorSplitRangeRequest> distributorSplitRange;
+	RequestStream<struct GetStorageWigglerStateRequest> storageWigglerState;
```

Comment:
This changes the protocol.

Reply:
Yeah... It requires more code to handle an Endpoint added in a patch release : (
I'm not sure how Apple thinks of this because basically @neethuhaneesha seems to want this info for RocksDB wiggle. In Snowflake, we decided not to patch and wait for 7.2.

Reply:
It's not clear to me how this would actually break anything.
- If new code gets a DD interface from a old code peer, the storageWigglerState request stream would be default initialized and just wouldn't function but it wouldn't crash.  
- If old code gets a DD interface from a new code peer, it would simply ignore the final request stream and the old code would not have any code trying to call it because it doesn't have the definition.  FlatBuffer deserialization will skip the extra member based, it doesn't need to know what the type is to do that.

Reply:
> If new code gets a DD interface from an old code peer, the storageWigglerState request stream would be default initialized and just wouldn't function but it wouldn't crash.

I think what Jingyu worries about is this case?

Reply:
This endpoint is only used by fdbserver, specifically the ClusterController to prepare a status result.  It would be very unusual to have a cluster of processes on different patch versions.

However, if FDB were to be deployed this way such that the cluster controller runs newer code than the data distributor, what _should_ happen is that the section of the status document that needs this endpoint is just missing and an error is added to the errors list in the status document.  However, I just checked the code and I think it might either wait forever, causing status to time out, or it might throw an error causing status to not be generated.  

@sfc-gh-xwang please address this with a new PR.  All remote requests related to the PW need to handle timeouts or errors as they can happen due to network hiccups or a well timed DD restart.  If you look above
https://github.com/apple/foundationdb/blob/9ca8a3c6837b41e16b6e7ae29d2c4a756aa95cff/fdbserver/Status.actor.cpp#L3103
at the other status section fetchers and their implementations, they all have timeout and error handling either in `clusterGetStatus()` directly or inside the `...Fetcher()` actor.  For example in
https://github.com/apple/foundationdb/blob/9ca8a3c6837b41e16b6e7ae29d2c4a756aa95cff/fdbserver/Status.actor.cpp#L3103
`timeoutError()` is used and there is a `catch` to output an error message if that or any other error occurs, which is why the fetchers normally take `&status_incomplete_reasons` as a parameter.

In `storageWigglerStatsFetcher()` the read loop would try forever.  Also, the `commit()` there on a read-only transaction is not necessary.  Also waiting on `stateFut` will not time out, it will wait forever for success or error and no error is output if the result is error.

Reply:
Good point! I think this is a low risk change, since `DataDistributorInterface` is not exposed to clients. I'll check if we really want this feature or not and then decide if we want to merge this PR.